caliban_toolbox/pre_annotation/data_loader.py

```diff
@@ -0,0 +1,462 @@
+        positions/FOVs (list): which positionss/FOVs to include
+
+            - list should be strings and match CellNet ontology
+              (e.g. data type = ['dynamic', '2d'])
+            - 'all' selects all data from a given catagory
+              (e.g. sessions=['all'])
+
+    Returns:
+        Numpy array with the shape [fovs, tifs, y_dim, x_dim]
+        Python dictionary containing metadata
+    """
+
+    def __init__(self,
+                 data_type,
+                 imaging_types,
+                 specimen_types,
+                 compartments=None,
+                 markers=['all'],  # the following should be sets to prevent double 'alls etc
+                 exp_ids=['all'],
+                 sessions=['all'],
+                 positions=['all'],
+                 file_type='.tif'):
+
+        if compartments is None and imaging_types != ['phase']:
+            raise ValueError('compartments is not specified')
+
+        self.data_type = set(data_type)
+        self.imaging_types = set(imaging_types)
+        self.specimen_types = set(specimen_types)
+        self.compartments = set(compartments)
+        self.markers = set(markers)
+        self.exp_ids = set(exp_ids)
+        self.sessions = set(sessions)
+        self.positions = set(positions)
+        self.file_type = file_type
+        self.onto_levels = np.full(7, False)
+
+        self._vocab_check()
+
+        self.base_path = '/data/raw_data'
+        for item in self.data_type:
+            self.base_path = os.path.join(self.base_path, item)
+
+        self._datasets_available()  # TODO: keep list of datasets for comparison
+        self._calc_upper_bound()
+
+        # self.mng_db = self._setup_mongo()
+
+    def _vocab_check(self):
+        """Check each user input for common mistakes and correct as neccesary
+        """
+        # TODO: improve this for generality and scale
+
+        # imaging_types - check for fluo misspellings
+        new_imaging_types = []
+        for item in self.imaging_types:
+            if any([item.lower() == 'flourescent',
+                    item.lower() == 'fluorescence',
+                    item.lower() == 'fluorescent',
+                    item.lower() == 'fluo']):
+                new_imaging_types.append('fluo')
+            elif item.lower() == 'phase':
+                new_imaging_types.append('phase')
+            else:
+                new_imaging_types.append(item)
+
+        self.imaging_types = new_imaging_types
+
+        # compartments - check for nuc or capitalization
+        new_compartments = []
+        for item in self.compartments:
+            if any([item.lower() == 'nuc',
+                    item.lower() == 'nuclear']):
+                new_compartments.append('Nuclear')
+            elif any([item.lower() == 'wholecell',
+                      item.lower() == 'whole_cell']):
+                new_compartments.append('WholeCell')
+            else:
+                new_compartments.append(item)
```

Comment:
Could clean this up by first setting `item = item.lower()`, then just checking if `item in {'nuc', 'nuclear'}`. A bit easier to extend in the future that way I think.

Reply:
done and added dicts to keep track of future misspellings